### PR TITLE
Allow int or str IDs as streaming arguments

### PR DESCRIPTION
**Pull request by fitnr**
Fix for #829. The str.join method raises a `ValueError` for `int`s.

### CODE REVIEW

**Comment by dannguyen** [comment]
Thanks @fitnr. I'm only just getting acquainted with using the Streaming API vs Tweepy so I know there are other dependencies and conventions to consider, so this is for future discussion: it might be worthwhile **to throw a warning** when a user passes in a string.

Right now, the behavior -- and I'm assuming this is unintended -- is to silently fail when a string value is passed into the `follow` argument/parameter. But because the `follow` parameter, according to Twitter's API spec, *only* accepts user IDs: 

https://dev.twitter.com/streaming/overview/request-parameters#follow

> A comma-separated list of user IDs, indicating the users whose Tweets should be delivered on the stream. 

In other words, it is never the case that the Twitter API will accept `follow=["ev"]`. Shouldn't the user be warned that what they are trying will *always* turn up nothing? 

In fact, I'd almost argue that it's better for `filter` to _throw an error_. Consider the uncommon case in which a user's **screen name** is entirely made up of numbers. Such as Twitter user [@101](https://twitter.com/101), whose __user_id__ is actually [3781411](http://gettwitterid.com/?user_name=101&submit=GET+USER+ID).

What behavior should the user expect with the following invocation?

       mystream.filter(follow=['101'])


Forcing the API of `filter` to accept *only* integers provides at least an implication of what the `follow` parameter is limited to:

       mystream.filter(follow=[101])

**Comment by fitnr** [comment]
I don't think changing signature of the `stream.filter` function to now throw warnings or errors when sequences of `str`s are passed is a friendly choice. Since a Twitter ID may, for good reason, be stored either as an `int` or a `str`, I say let the [function accept either](https://docs.python.org/3.5/glossary.html#term-duck-typing), and throw an error if Twitter complains. Designing guard rails or trying to police a certain kind of behvaior seems unnecessary in a library that  tracks a complicated and ever-changing API.

To answer your question about this:
```python
mystream.filter(follow=['101'])
````
I would expect that will produce a stream that follows the user with `user_id == 101`. I would base that on reading the [API spec on streaming](https://dev.twitter.com/streaming/overview/request-parameters#follow). It should also be documented in the Tweepy docs (which are currently a little garbled on the streaming API, but that's another issue).

File: tweepy/streaming.py [hunk 1]
```diff
@@ -430,10 +430,13 @@ def filter(self, follow=None, track=None, async=False, locations=None,
         if self.running:
             raise TweepError('Stream object already connected!')
         self.url = '/%s/statuses/filter.json' % STREAM_VERSION
+
         if follow:
-            self.body['follow'] = u','.join(follow).encode(encoding)
+            self.body['follow'] = u','.join(str(f) for f in follow).encode(encoding)
```
Comment (Harmon758) [inline tweepy/streaming.py:435]:
```suggestion
            self.body['follow'] = u','.join(map(str, follow)).encode(encoding)
```

File: tweepy/streaming.py [hunk 2]
```diff
@@ -430,10 +430,13 @@ def filter(self, follow=None, track=None, async=False, locations=None,
...
+
         if track:
-            self.body['track'] = u','.join(track).encode(encoding)
+            self.body['track'] = u','.join(str(f) for f in track).encode(encoding)
```
Comment (Harmon758) [inline tweepy/streaming.py:438]:
```suggestion
            self.body['track'] = u','.join(map(str, track)).encode(encoding)
```

**Review by Harmon758 — CHANGES_REQUESTED** [review]

**Review by Harmon758 — APPROVED** [review]

**Pull request merged**